### PR TITLE
refactor(navigation-bar): update styled system navigation bar (FE-3560)

### PR DESCRIPTION
**Pull request by nicktitchmarsh**
Fixes FE-3560

### Proposed behaviour
Remove inappropriate props that should not be available in navigation bar.

Media queries and breakpoints were also inconsistent with DS docs

### Current behaviour
Box props and margins are available where they shouldn't

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/navigation-bar/navigation-bar.component.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const NavigationBar = ({
 NavigationBar.propTypes = {
   /** Styled system spacing props */
   ...propTypes.space,
```
Comment (edleeks87) [inline src/components/navigation-bar/navigation-bar.component.js:28]: We should filter the the prop types here using the `filterStyledSystemPaddingProps` so if a consumer uses any `margin` props they will get the prop type warning. 

https://github.com/Sage/carbon/blob/a99d9679e2a13f789976f3c7bcd02bd411301ca5/src/components/popover-container/popover-container.component.js#L15

File: src/components/navigation-bar/navigation-bar.style.js [hunk 2]
```diff
@@ -1,13 +1,11 @@
 import styled, { css } from "styled-components";
-import { space } from "styled-system";
+import { padding, flexbox } from "styled-system";
 import { baseTheme } from "../../style/themes";
-import Box from "../box";
 
-const StyledNavigationBar = styled(Box).attrs({ as: "nav" })`
-  padding: 0 40px;
+const StyledNavigationBar = styled.div.attrs({ as: "nav" })`
```
Comment (edleeks87) [inline src/components/navigation-bar/navigation-bar.style.js:5]: I think this could just be

```suggestion
const StyledNavigationBar = styled.nav`
```

**Review by sjohnson-sage — COMMENTED** [review]
Nothing more to add, I'll approve once you've sorted Ed's comments 👍🏻

**Review by edleeks87 — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**